backend/telemetry_shard/src/main.rs

```diff
@@ -283,21 +289,28 @@ where
                 let message_id = node_message.id();
                 let payload = node_message.into_payload();
 
-                // Ignore messages from IDs that exceed our limit:
-                if message_ids_seen.contains(&message_id) {
-                    // continue on; we're happy
-                } else if message_ids_seen.len() >= max_nodes_per_connection {
-                    // ignore this message; it's not a "seen" ID and we've hit our limit.
-                    continue;
-                } else {
-                    // not seen ID, not hit limit; make note of new ID
-                    message_ids_seen.insert(message_id);
-                }
-
                 // Until the aggregator receives an `Add` message, which we can create once
                 // we see one of these SystemConnected ones, it will ignore messages with
                 // the corresponding message_id.
                 if let node_message::Payload::SystemConnected(info) = payload {
+                    let network_id = info.node.network_id;
+
+                    // Node with network ID already exist? Remove it to be replaced with this one.
+                    if let Some(&old_message_id) = network_id_to_msg_id.get(&network_id) {
+                        network_id_to_msg_id.remove(&network_id);
+                        allowed_message_ids.remove(&old_message_id);
+                        let _ = tx_to_aggregator.send(FromWebsocket::Remove { message_id: old_message_id }).await;
+                    }
+                    // Too many nodes seen on this connection? Ignore this one.
+                    else if network_id_to_msg_id.len() >= max_nodes_per_connection {
+                        continue;
```

Comment:
Should we collect stats on this so we know if/when things go haywire?

Reply:
I added a log message here at least; collecting stats on it would be a little trickier at the mo!